### DISCUSSION TITLE
how to mount two sub app with its own static files?

### DISCUSSION THREAD

**Discussion by phoenixor** [fastapi/fastapi#6099] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles

app = FastAPI()
subapp1= FastAPI()
subapp2 = FastAPI()

app.mount("/subapp1", subapp1)
app.mount("/subapp2", subapp2)

subapp1.mount(
    "/assets",
    StaticFiles(directory="app/subapp1/frontend/dist/assets"),
)
subapp2.mount(
    "/assets",
    StaticFiles(directory="app/subapp2/frontend/dist/assets"),
)

@subapp1.get("/")
async def subapp1():
    return FileResponse("app/subapp1/frontend/dist/index.html")
@subapp2.get("/")
async def subapp2():
    return FileResponse("app/subapp2/frontend/dist/index.html")


if __name__ == "__main__":
    import uvicorn
    uvicorn.run("main:app", host="0.0.0.0", port=8080, reload=True)
```

### Description

when calling "http://localhost:8080/subapp1/" and "http://localhost:8080/subapp2/"
only subapp1 success, but subapp2 failed with static files 404

### Operating System

Windows

### FastAPI Version

0.88.0

### Python Version

3.10.2

**Comment by phoenixor** [comment]
i have solve this problem, its a matter of the frontend

**Reply by mdo6180** [reply]
can i see how you resolved this issue?

**Reply by mdo6180** [reply]
i don't know if this is an answer, but this works just fine for me.
```python
from fastapi import FastAPI
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles


newline = "\n"

app1 = FastAPI()
app1.mount("/static", StaticFiles(directory="./static1"), name="static")

app2 = FastAPI()
app2.mount("/static", StaticFiles(directory="./static2"), name="static")


app = FastAPI()
app.mount("/app1", app1)
app.mount("/app2", app2)
app.mount("/static", StaticFiles(directory="./static"), name="static")

@app.get("/", response_class=HTMLResponse)
async def home():
    return f"""
    <!DOCTYPE html>
    <html>
        <head>
            <meta charset="UTF-8">
            <title>Multiple StaticFiles</title>

            <!-- Static files for app1 -->
            <link rel="stylesheet" href="/app1/static/style.css">

            <!-- Static files for app2 -->
            <link rel="stylesheet" href="/app2/static/style.css">

            <!-- Main static files -->
            <link rel="stylesheet" href="/static/style.css">
        </head>
        <body>
            <div id="app1">App 1</div>
            <div id="app2">App 2</div>
            <div id="main">Main</div>
        </body>
    </html> 
    """
```

**Comment by NuISheRa78964** [comment]
> i have solve this problem, its a matter of the frontend

Hi Can you show u's how you resolved the issue. Tks
